### DISCUSSION TITLE
role_has_permissions too slow

### DISCUSSION THREAD

**Discussion by princepeal** [spatie/laravel-permission#2627] [Q&A]
I am using this version and Laravel 10

`"spatie/laravel-permission": "^5.5",`

I have 200 permissions it's a larve application so we need many permissions here and there. 

```
$permissions = "fetch permissions here";

$role->givePermissionTo($permissions[$x]); //only this line making it slow 
```

Thanks

**Comment by parallels999** [comment]
try V6
and `givePermissionTo` it's only when permissions are assigned, it creates the cache, later just `->can` ot `->hasRole` are quick
